### ISSUE TITLE
will count as bug?

### ISSUE THREAD

**Issue by abchugh** [slatex/sTeX-React#130]
A content SUGGESTION was logged by "Akash Yadav" at the following url:

https://courses.voll-ki.fau.de/course-notes/ai-1?inDocPath=-z0bd29%7Ef98e019f

## The issue as described by the user:
```
when hover on Statistical AI definition, it is not marking it as visited, so if again Statistical AI appears in definition then it will open same definition again, which is a bad recursion/loop. 
Either this scenario has been left out or not considered. This can fix by using "visit only once Graph" Algorithm.
```

## The text highlighted while reporting this issue:
```
statistical approaches
```

### The selected text was in the following section hierarchy:

**_INNERMOST SECTION FIRST_**

1. GitLab: https://gl.mathhub.info/MiKoMH/AI/-/blob/main/source/course/sec/concept.en.tex<br/>FetchURL: https://stexmmt.mathhub.info//:sTeX/fulldocument?archive=MiKoMH/AI&filepath=course/sec/concept.en.xhtml&bindings=0_0_l

2. GitLab: https://gl.mathhub.info/MiKoMH/AI/-/blob/main/source/course/sec/pre.en.tex<br/>FetchURL: https://stexmmt.mathhub.info//:sTeX/fulldocument?archive=MiKoMH/AI&filepath=course/sec/pre.en.xhtml&bindings=0_0_5

3. GitLab: https://gl.mathhub.info/MiKoMH/AI/-/blob/main/source/course/notes/notes1.tex<br/>FetchURL: https://stexmmt.mathhub.info//:sTeX/fulldocument?archive=MiKoMH/AI&filepath=course/notes/notes1.xhtml

**Comment by kohlhase** [comment]
You are right this is confusing behavior in the UI. I am not sure that I would count this as a bug (but I will classify this issue as one anyway), but I agree that this should be changed. E.g. by a hover that indicates that there no new information is to be had. I agree on the algorithm, but I am not sure what the UI should be. 
Asigning to  @abchugh

**Comment by abchugh** [comment]
I, for one, find the recursion very amusing. But my amusement shouldn't come in the way of user experience. 

I feel that the core problem is that the definition of statistical AI references the concept itself (See https://gl.mathhub.info/smglom/ai/-/blob/main/source/mod/statistical-AI.en.tex). 

I feel that authors of definitions should not do that. What do you think about that.

And if think this kind of self-reference is correct/justified, RusTeX/MMT should convert it into a pink link and not a blue link.

I can hack this on the frontend but it doesn't feel like the right place.

**Comment by kohlhase** [comment]
* I believe we cannot reasonably get around having definitions reference the definienda in the definitions. 
* No, we should not re-color the reference, as the reference is not a definiendum, that would confuse students even more. 
* We should probably have a better way of telling the students "this concept is defined here already" here than the endless recursion.

**Comment by abchugh** [comment]
@Jazzpirate Is this resolved in ALeA4?
